lib/sudo-exec/src/monitor.rs

```diff
@@ -42,7 +42,11 @@ impl MonitorClosure {
 
             // Wait for the main sudo process to give us green light before spawning the command. This
             // avoids race conditions when the command exits quickly.
-            let MonitorEvent::ExecCommand = retry_while_interrupted(|| backchannel.recv())?;
+            let event = retry_while_interrupted(|| backchannel.recv())?;
+
+            // FIXME: ogsudo doesn't check that this event is not a forwarded signal from the
+            // parent process. What should we do in that case?
+            assert_eq!(event, MonitorEvent::ExecCommand);
```

Comment:
can that happen at this point in time? the `ParentClosure` only sends `MonitorEvent::Signal`s after sending this first `MonitorEvent::ExecCommand`, right? and the `UnixStream::pair` delivers messages in order, I assume

Reply:
Afaik this type of socket should respect order, and the signal callbacks will only be called once `event_loop` is running, so yeah we shouldn't need this assertion. I'd leave it as a debug assertion just in case.